base/printf.jl

```diff
@@ -1183,7 +1187,7 @@ function _printf(macroname, io, fmt, args)
           quote
              G = $(esc(x))
              if length(G) != $(length(sym_args))
-                throw(ArgumentError($macroname,": wrong number of arguments (",length(G),") should be (",$(length(sym_args)),")"))
+                 throw(ArgumentError(string($macroname,": wrong number of arguments (",length(G),") should be (",$(length(sym_args)),")")))
```

Comment:
Usually we don't mention the function name in the message, perhaps just change this to
```
throw(ArgumentError(": wrong number of arguments (",length(G),") should be (",$(length(sym_args)),")"))
```
?

Reply:
The same kind of error messages appears 3 times in the `_printf` function. Should I change them all?

Reply:
Ah, perhapse its there for a reason then since both `@sprintf` and `@printf` call the same method.

Reply:
The function is called form `@sprintf`and `@printf`. But that is also visible in the stack trace. I don't see more reasons than everywhere else to put the name into the exception string.

Reply:
> that is also visible in the stack trace. <
Unfortunately not always true:
```
julia> @sprintf "%d" 1:2...
ERROR: ArgumentError: : wrong number of arguments (2) should be (1)
Stacktrace:
 [1] top-level scope at printf.jl:1190

julia> @sprintf "%d" 1 2
ERROR: LoadError: ArgumentError: wrong number of arguments (2) should be (1)
Stacktrace:
 [1] _printf(::String, ::Expr, ::String, ::Tuple{Int64,Int64}) at ./printf.jl:1168
 [2] @sprintf(::LineNumberNode, ::Module, ::Vararg{Any,N} where N) at /home/julia/julia/usr/share/julia/site/v0.7/Printf/src/Printf.jl:64
in expression starting at REPL[7]:1
```
That means, the macro name appears in the stack trace, if the exception occurs at macro expansion time, but not if it appears when the generated code is executed. That means, it would make sense to keep it in the line, I fixed, and to remove it at the other 2 spots.

Reply:
Lets just keep it as is.

Reply:
ok - then it can be merged as is.